### PR TITLE
ETFE-2553 - CAM-ITM38: commodity-code

### PR DESCRIPTION
**Pull request by kabuumu**

### CODE REVIEW

File: app/viewmodels/checkAnswers/sections/items/ItemCommodityCodeSummary.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.SummaryListRow
+import viewmodels.checkAnswers.sections.items.ItemBrandNameSummary.getValue
+import viewmodels.govuk.summarylist._
+import viewmodels.implicits._
+
+object ItemCommodityCodeSummary {
+  def row(idx: Index)(implicit request: DataRequest[_], messages: Messages): Option[SummaryListRow] = {
+    request.userAnswers.get(ItemCommodityCodePage(idx)).map { value =>
+      SummaryListRowViewModel(
+        key = "itemCommodityCode.checkYourAnswersLabel",
```
Comment (r-melvin) [inline app/viewmodels/checkAnswers/sections/items/ItemCommodityCodeSummary.scala:36]: Doesn't this message take a parameter?

File: test/navigation/ItemsNavigatorSpec.scala [hunk 2]
```diff
@@ -36,28 +36,13 @@ class ItemsNavigatorSpec extends SpecBase {
...
-            s"when the EPC is $epc" in {
-              navigator.nextPage(
-                ItemExciseProductCodePage(testIndex1),
-                NormalMode,
-                emptyUserAnswers.set(ItemExciseProductCodePage(testIndex1), epc)) mustBe
-                testOnly.controllers.routes.UnderConstructionController.onPageLoad()
-            }
-          })
-        }
-
```
Comment (r-melvin) [inline test/navigation/ItemsNavigatorSpec.scala:52]: This is for a different page, I'm not sure this should be deleted

File: test/navigation/ItemsNavigatorSpec.scala [hunk 3]
```diff
@@ -347,28 +332,13 @@ class ItemsNavigatorSpec extends SpecBase {
...
-            s"when the EPC is $epc" in {
-              navigator.nextPage(
-                ItemExciseProductCodePage(testIndex1),
-                CheckMode,
-                emptyUserAnswers.set(ItemExciseProductCodePage(testIndex1), epc)) mustBe
-                testOnly.controllers.routes.UnderConstructionController.onPageLoad()
-            }
-          })
-        }
-
```
Comment (r-melvin) [inline test/navigation/ItemsNavigatorSpec.scala:363]: as above

File: app/navigation/ItemsNavigator.scala [hunk 4]
```diff
@@ -117,12 +117,8 @@ class ItemsNavigator @Inject() extends BaseNavigator {
 
   private def epcRouting(idx: Index, userAnswers: UserAnswers, mode: Mode): Call =
     userAnswers.get(ItemExciseProductCodePage(idx)) match {
-      case Some("S500" | "T300" | "S400" | "E600" | "E800" | "E910") =>
-        //TODO: Route to CAM-ITM43 when implemented
-        testOnly.controllers.routes.UnderConstructionController.onPageLoad()
       case Some(_) =>
```
[the remark this answers is not in the document]
Comment (james-work-account) [reply app/navigation/ItemsNavigator.scala:120]: @r-melvin, we have to make that reference data "get relevant CN Code(s)" call even for these EPCs. The ItemCommodityCodeController.onPageLoad route:

1. Calls out to reference data
2. If an EPC has >1 CN Code, render the page
3. If an EPC has <=1, save that code and redirect

AKA,
image

File: test/navigation/ItemsNavigatorSpec.scala (hunk 2 above)
Comment (james-work-account) [reply test/navigation/ItemsNavigatorSpec.scala:52]: Related to my comment above - these EPCs shouldn't be treated differently as far as the navigator is concerned.

File: test/navigation/ItemsNavigatorSpec.scala (hunk 3 above)
Comment (james-work-account) [reply test/navigation/ItemsNavigatorSpec.scala:363]: Related to my comment above - these EPCs shouldn't be treated differently as far as the navigator is concerned.

File: app/navigation/ItemsNavigator.scala (hunk 4 above)
[the remark this answers is not in the document]
Comment (r-melvin) [reply app/navigation/ItemsNavigator.scala:120]: Ok that's fair enough but now we have navigation logic in multiple places so won't that make it a little more confusing to maintain?

**Review by r-melvin — APPROVED** [review]

**Review by james-work-account — APPROVED** [review]

**Pull request merged**